### ISSUE TITLE
Error building MUMPS

### ISSUE THREAD

**Issue by wrs28** [JuliaSmoothOptimizers/MUMPS.jl#43]
Since `brew upgrade`, MUMPS no longer builds in Julia.

tl;dr: update to openblas0.3.6 in homebrew seems not to have propagated to brewsci-mumps formula, breaking the `Pkg.build("MUMPS")`

The error message from the build log is
```
ERROR: LoadError: unable to locate libmumps_common.dylib... please install MUMPS or set MUMPS_PREFIX
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] top-level scope at /Users/wrs/.julia/packages/MUMPS/6EV6j/deps/build.jl:29
 [3] include at ./boot.jl:326 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include(::Module, ::String) at ./sysimg.jl:29
 [6] include(::String) at ./client.jl:403
 [7] top-level scope at none:0
in expression starting at /Users/wrs/.julia/packages/MUMPS/6EV6j/deps/build.jl:23
```

I investigated `build.jl` and identified where the error occurred, which is in loading `libmumps_common.dylib`.

If I directly try to load the library in the REPL using `Libdl.dlopen` I get 
```
ERROR: could not load library "/usr/local/opt/brewsci-mumps/lib/libmumps_comon.dylib"
dlopen(/usr/local/opt/brewsci-mumps/lib/libmumps_comon.dylib, 1): image not found
```

which is not terribly informative.

On the other hand, trying to load any of the other dylibs, such as `libdmumps.dylib` or the crucial `libmumps_simple.dylib` I get the error message
```
ERROR: could not load library "/usr/local/opt/brewsci-mumps/lib/libdmumps.dylib"
dlopen(/usr/local/opt/brewsci-mumps/lib/libdmumps.dylib, 1): Library not loaded: /usr/local/opt/openblas/lib/libopenblasp-r0.3.5.dylib
  Referenced from: /usr/local/opt/brewsci-mumps/lib/libdmumps.dylib
  Reason: image not found
```

The current openblas version from brew is v0.3.6 (which was released about 10 days ago), while it seems libZmumps is looking for v0.3.5.

It seems that the source of the error is in the build from Homebrew, but I'm not sure where to report that or how to fix it, so I'm reporting it here instead.

**Comment by dpo** [comment]
Thanks for the report! The reason is almost surely that `brewsci-mumps` and its dependencies need to be rebuilt over at https://github.com/brewsci/homebrew-num. I'm tied up for the next week or so. Feel free to open pull requests over there. Otherwise, you could also `brew install -s brewsci-mumps` to compile MUMPS locally.

**Comment by dpo** [comment]
> ERROR: could not load library "/usr/local/opt/brewsci-mumps/lib/libmumps_comon.dylib"
dlopen(/usr/local/opt/brewsci-mumps/lib/libmumps_comon.dylib, 1): image not found

You seem to have a typo here: "comon" instead of "common". You'll probably get the same OpenBLAS error message anyways.

**Comment by wrs28** [comment]
Haha yes good catch.

Indeed I get the same OpenBLAS error when I take the time to spell "common" correctly.

I'll try building locally, thanks for the suggestion!

**Comment by wrs28** [comment]
Hmm it seems that building locally did not help, even after removing old brewsci-mumps builds (and dependencies) and cleaning up.

I'm sure it will get sorted out at Brewsci, but I'll open an issue there in the meantime for completeness.

Thanks for the help!

**Comment by wrs28** [comment]
Recent `brew upgrade` fixed the problem. Thanks all.
